### ISSUE TITLE
Fix Request: false positive AvoidUnguardedMutableFieldsInSharedObjects and AvoidUnguardedAssignmentToNonFinalFieldsInSharedObjects for Prototype scope beans

### ISSUE THREAD

**Issue by hem-verma** [jborgers/PMD-jPinpoint-rules#147]
We are getting AvoidUnguardedMutableFieldsInSharedObjects and AvoidUnguardedAssignmentToNonFinalFieldsInSharedObjects for Prototype scope bean. The idea of prototype scope is that it returns new bean from context every time it is requested. so there is no issue of thread safety. Rule should be updated to check for prototype scope.

```
@Scope(value = ConfigurableBeanFactory.SCOPE_PROTOTYPE)
@Component
public class ProductBuilder {
    private String id;
    private String productTypeCode;
    private String administrationCode;
    private String name;
    private ProductIdentifier identifier;

... //rest of the builder code
}
```
